assets/js/base/hocs/with-reviews.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { speak } from '@wordpress/a11y';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { getReviews } from '../../blocks/reviews/utils';
+
+const withReviews = ( OriginalComponent ) => {
+	class WrappedComponent extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.state = {
+				error: null,
+				loading: false,
+				reviews: [],
+				totalReviews: 0,
+			};
+
+			this.setError = this.setError.bind( this );
+			this.delayedAppendReviews = this.props.delayFunction( this.appendReviews, 400 );
+		}
+
+		componentDidMount() {
+			this.replaceReviews();
+		}
+
+		componentDidUpdate( prevProps ) {
+			if ( prevProps.reviewsToDisplay < this.props.reviewsToDisplay ) {
+				// Since this attribute is controlled with a slider,
+				// it's better not to load the reviews immediately.
+				this.delayedAppendReviews();
+			} else if (
+				prevProps.categoryIds !== this.props.categoryIds ||
```

Comment:
I refactored this a bit, `@wordpress/is-shallow-equal` would add another dependency so I tried to avoid it. Feel free to make any suggestions. :slightly_smiling_face: